patch.yml

```diff
@@ -13,6 +13,36 @@
 # limitations under the License.
 
 ---
+- name: Play | Set correct RDBMS release and set fact if this is a single node inventory
```

Comment:
Please accept my apologies if I commented on this in the past, but logic like getting the highest version is typically put into files under roles/, keeping the top-level patch.yml file very short.  Is it possible to take this logic (which will likely no longer need an explicit inclusion of defaults) to /roles/patch/tasks/main.yml?  Thanks!

Reply:
Thanks for all your suggestions so far with the review that helps to bolster the quality of the code finally checked in.

On this particular point:
`roles/patch/tasks/main.yml` is not the sole entry point to the tasks from `patch.yml`.
The resolution for the variable `oracle_rel` is needed at this this top-level `patch.yml`, because other playboooks like main.yml, rac-opatch.yml, etc are called from this top-level `patch.yml` and they all consume the variable set in their `when` clauses.

If we have to move, it will be code repetition in those individual playbooks, as the logic depends on the correct of this.

Reply:
My reading of https://docs.ansible.com/ansible/latest/collections/ansible/builtin/set_fact_module.html 👍 

```
The set_fact module takes ``key=value`` pairs or ``key: value``(YAML notation) as variables to set in the playbook scope. 
```

My erading is that setting a fact will leave it set for the entire playbook.  Would you mind testing to confirm?

Reply:
Happy to test, but it's not point I was trying to make.
Please allow me to reword my response:

To do what you are saying, we will have to copy paste the exact block in multiple playbooks under the roles/patch/tasks/ directory.
The reason is the codeflow path from patch.yml differs whether one is patching a standalone system or a RAC system.

Reply:
Discussed this offline, so I think this comment is moot now.